roxctl/common/tls.go

```diff
@@ -48,26 +48,26 @@ func (v *insecureVerifierWithWarning) VerifyPeerCertificate(leaf *x509.Certifica
 }
 
 // ConnectNames returns the endpoint and (SNI) server name given by the
-// --endpoint and --server-name flags respectively. If no server name is given,
-// an appropriate name is derived from the given endpoint.
-func ConnectNames() (string, string, error) {
-	endpoint, _, err := flags.EndpointAndPlaintextSetting()
+// --endpoint and --server-name flags respectively and information about plaintext.
+// If no server name is given, an appropriate name is derived from the given endpoint.
+func ConnectNames() (string, string, bool, error) {
+	endpoint, usePlaintext, err := flags.EndpointAndPlaintextSetting()
 	if err != nil {
-		return "", "", errors.Wrap(err, "could not get endpoint")
+		return "", "", false, errors.Wrap(err, "could not get endpoint")
 	}
 	serverName := flags.ServerName()
 	if serverName == "" {
 		var err error
 		serverName, _, _, err = netutil.ParseEndpoint(endpoint)
 		if err != nil {
-			return "", "", errors.Wrap(err, "could not parse endpoint")
+			return "", "", false, errors.Wrap(err, "could not parse endpoint")
 		}
 	}
-	return endpoint, serverName, nil
+	return endpoint, serverName, usePlaintext, nil
```

Comment:
nit refactoring (due to two functions of the function, and therefore occasionally unused return values):
1. extract `endpoint, usePlaintext, err := flags.EndpointAndPlaintextSetting()`;
2. make `endpoint` the function parameter;
3. rename the function to `getServerName(endpoint string) (string, error)`.

Reply:
I don't get it. You would like to extract function that will be only used in this single place?

Reply:
No, move it out of the function (to the caller) and pass the result as the argument.